### PR TITLE
solution

### PR DESCRIPTION
**Pull request by NikitaKarbazin**
[DEMO LINK](https://nikitakarbazin.github.io/react_redux-list-of-todos/)

### CODE REVIEW

**Review by vsupruniuk — APPROVED** [review]
👍

File: src/store/action-creators/todos.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      const response = await getTodos();
+
+      dispatch({
+        type: TodosActionTypes.FETCH_TODOS_SUCCESS,
+        payload: response,
+      });
+    } catch {
+      dispatch({
+        type: TodosActionTypes.FETCH_TODOS_ERROR,
+        payload: 'Something happen on the Server',
```
[the remark this answers is not in the document]
Comment (mykola-koval) [reply src/store/action-creators/todos.ts:19]: and why are you dispatching when there is an error? maybe let's do something else)

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,83 @@
...
-      <Start title="Start loading" />
-      <Finish title="Succeed loading" message="Loaded successfully!" />
-      <Finish
-        title="Fail loading"
-        message="An error occurred when loading data."
-      />
+      <h1 className="title">Redux list of todos</h1>
+      {loading && <Loader />}
+      {loading && error && (
+        <h2>{error}</h2>
```
Comment (mykola-koval) [inline src/App.tsx:61]: when there is an error after loading will it be shown ?
when there is an error will it be shown with a loader together?

File: src/App.tsx [hunk 3]
```diff
@@ -1,26 +1,83 @@
...
+      )}
+      {!loading && error === null && (
```
Comment (mykola-koval) [inline src/App.tsx:63]:
```suggestion
      {!loading && !error && (
```

File: src/components/TodoModal/TodoModal.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+          <div className="modal-card-body">
+            <p className="block">{todo?.title}</p>
+
+            <p className="block">
+              {todo?.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
```
Comment (mykola-koval) [inline src/components/TodoModal/TodoModal.tsx:58]: i am not sure but i think it should work that way
```suggestion
              {todo?.completed 
              ? <strong className="has-text-success">Done</strong>
              : <strong className="has-text-danger">Planned</strong>
              }
```

File: src/components/TodoList.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+  const [start, finish] = updateNextPieceOfData(
+    total,
+    page,
+    amountItemsPerPage,
+  );
+
+  const visibleTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case 'all':
```
Comment (mykola-koval) [inline src/components/TodoList.tsx:31]: let's use enum here

**Review by mykola-koval — APPROVED** [review]
gj 👍

**Pull request closed without merging**